src/test/javascript/acd/ol/ol.mapwithactions.test.js

```diff
@@ -1,188 +1,236 @@
+
+        spyOn(action1, 'activate');
+        spyOn(action2, 'activate');
+        spyOn(action2, 'deactivate');
+
+        map.activateDefaultAction();
+
+        afterActivation(function () {
+          expect(action2.deactivate).toHaveBeenCalled();
+          expect(action1.activate).toHaveBeenCalled();
+        }, done);
+      });
+
+  it('Wanneer de default actie wordt geactiveerd, zal de huidige actie gedeactiveerd worden en is de nieuwe default actie de eerst gedefinieerde actie, ook als de huidige actie de eerst gedefinieerde is',
+      function (done) {
+        var map = createMapWithActions();
+
+        map.activateAction(action1);
+        expect(map.currentAction).toBe(action1);
+
+        spyOn(action1, 'activate');
+        spyOn(action1, 'deactivate');
+
+        map.activateDefaultAction();
+
+        afterActivation(function () {
+          expect(action1.deactivate).toHaveBeenCalled();
+          expect(action1.activate).toHaveBeenCalled();
+        }, done);
+      });
+
+  it('bij het aanmaken van een kaart met acties wordt standaard functionaliteit toegevoegd aan de kaart dat bij escape de eerste kaart actie geactiveerd wordt',
+      function () {
+        var map = new acd.ol.MapWithActions({
+          actions: []
+        });
+        spyOn(map, 'activateDefaultAction');
+        expect(map.activateDefaultAction).not.toHaveBeenCalled();
+        var event = new Event('keydown');
+        event.keyCode = 27;
+        document.body.dispatchEvent(event);
+        expect(map.activateDefaultAction).toHaveBeenCalled();
+      });
+
+  it('indien gewenst kan de standaard escape functionaliteit uitgeschakeld worden bij het aanmaken van een kaart met acties',
+      function () {
+        var map = new acd.ol.MapWithActions({
+          actions: [],
+          disableEscapeKey: true
+        });
+        spyOn(map, 'activateDefaultAction');
+        expect(map.activateDefaultAction).not.toHaveBeenCalled();
+        var event = new Event('keydown');
+        event.keyCode = 27;
+        document.body.dispatchEvent(event);
+        expect(map.activateDefaultAction).not.toHaveBeenCalled();
+      });
+
+  it('er zijn 9 predefined interactions', function () {
+    spyOn(ol.Map, 'call').and.callFake(function (map, options) {
+      expect(options.interactions.getLength()).toBe(9);//Standaard zijn er 9 interactions
+    });
+    var map = new acd.ol.MapWithActions({
+      actions: []
+    });
+    expect(ol.Map.call).toHaveBeenCalledTimes(1);
+  });
+
+  it('indien gewenst kan de standaard rotation functionaliteit uitgeschaked worden bij het aanmaken van een kaart met acties',
+      function () {
+        spyOn(ol.Map, 'call').and.callFake(function (map, options) {
+          expect(options.interactions.getLength()).toBe(7);//9-2=7
+        });
+        var map = new acd.ol.MapWithActions({
+          actions: [],
+          disableRotation: true
+        });
+        expect(ol.Map.call).toHaveBeenCalledTimes(1);
+  });
```

Comment:
Dit controleert alleen dat er minder interactions zijn, maar de test op zich controleert niet dat de rotation uitgeschakeld is.